protocol/messenger_filter_init.go

```diff
@@ -0,0 +1,351 @@
+	filtersCh := make(chan []transport.FiltersToInitialize, 3)
+	publicKeysCh := make(chan []*ecdsa.PublicKey, 2)
+
+	wg.Add(5)
+	go m.processJoinedCommunities(&wg, filtersCh, errCh)
+	go m.processSpectatedCommunities(&wg, filtersCh, errCh)
+	go m.processChats(&wg, filtersCh, publicKeysCh, errCh)
+	go m.processContacts(&wg, publicKeysCh, errCh)
+	go m.processControlledCommunities(&wg, errCh)
+
+	wg.Wait()
+	close(filtersCh)
+	close(publicKeysCh)
+	close(errCh)
+
+	return m.collectResults(filtersCh, publicKeysCh, errCh)
+}
+
+func (m *Messenger) processJoinedCommunities(wg *sync.WaitGroup, filtersCh chan<- []transport.FiltersToInitialize, errCh chan<- error) {
+	defer gocommon.LogOnPanic()
+	defer wg.Done()
+
+	joinedCommunities, err := m.communitiesManager.Joined()
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	filtersToInit := m.processCommunitiesSettings(joinedCommunities)
+	filtersCh <- filtersToInit
+}
+
+func (m *Messenger) processCommunitiesSettings(communities []*communities.Community) []transport.FiltersToInitialize {
+	logger := m.logger.With(zap.String("site", "processCommunitiesSettings"))
+	filtersToInit := make([]transport.FiltersToInitialize, 0, len(communities))
+
+	for _, org := range communities {
+		// the org advertise on the public topic derived by the pk
+		filtersToInit = append(filtersToInit, m.DefaultFilters(org)...)
+
+		if err := m.communitiesManager.EnsureCommunitySettings(org); err != nil {
+			logger.Warn("failed to process community settings", zap.Error(err))
+		}
+	}
+
+	return filtersToInit
+}
+
+func (m *Messenger) processSpectatedCommunities(wg *sync.WaitGroup, filtersCh chan<- []transport.FiltersToInitialize, errCh chan<- error) {
+	defer gocommon.LogOnPanic()
+	defer wg.Done()
+
+	spectatedCommunities, err := m.communitiesManager.Spectated()
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	filtersToInit := make([]transport.FiltersToInitialize, 0, len(spectatedCommunities))
+	for _, org := range spectatedCommunities {
+		filtersToInit = append(filtersToInit, m.DefaultFilters(org)...)
+	}
+	filtersCh <- filtersToInit
+}
+
+func (m *Messenger) processChats(wg *sync.WaitGroup, filtersCh chan<- []transport.FiltersToInitialize, publicKeysCh chan<- []*ecdsa.PublicKey, errCh chan<- error) {
+	defer gocommon.LogOnPanic()
+	defer wg.Done()
+
+	// Get chat IDs and public keys from the existing chats.
+	// TODO: Get only active chats by the query.
+	chats, err := m.persistence.Chats()
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	validChats, communityInfo := m.validateAndInitChatsFirstMessageTimestamp(chats)
+	filters, publicKeys, err := m.processValidChats(validChats, communityInfo)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	filtersCh <- filters
+	publicKeysCh <- publicKeys
+
+	if err := m.processDeprecatedChats(); err != nil {
+		errCh <- err
+	}
+}
+
+func (m *Messenger) validateAndInitChatsFirstMessageTimestamp(chats []*Chat) ([]*Chat, map[string]*communities.Community) {
```

Comment:
oh cmon 😄 Please avoid this `and` word in the functions name. It's a sign of smth wrong. 

And in this case it's very simple to remove: 
just call `initChatsFirstMessageTimestamp` _outside_ this function, right after the call to it.

```
validChats, communitiesCache := m.valiateChats(communitiesCache, validChats)
initChatsFirstMessageTimestamp(validChats, communitiesCache)
```